release-notes/10.0/preview/preview6/aspnetcore.md

```diff
@@ -0,0 +1,239 @@
+
+ASP.NET Core updates in .NET 10:
+
+- [What's new in ASP.NET Core in .NET 10](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-10.0) documentation.
+- [Breaking changes](https://docs.microsoft.com/dotnet/core/compatibility/10.0#aspnet-core)
+- [Roadmap](https://github.com/dotnet/aspnetcore/issues/59443)
+
+## Automatic eviction from memory pool
+
+The memory pools used by Kestrel, IIS, and HTTP.sys now automatically evict memory blocks when the application is idle or under less load, helping applications use resources more efficiently.
+
+**Why it matters:**
+Previously, memory allocated by the pool would remain reserved, even when not in use. With this enhancement, when the app is idle for a period of time, memory is now released back to the system, reducing overall memory usage and helping applications stay responsive under varying workloads.
+
+**How to use:**
+No action is needed to benefit from this feature. Memory eviction is handled automatically by the framework.
+
+There are also metrics added to the default memory pool used by our server implementations. The new metrics are under the name `Microsoft.AspNetCore.MemoryPool`. See the [ASP.NET Core metrics documentation](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics) for general information on what metrics are and how to use them.
+
+You can also use the new `IMemoryPoolFactory` service interface to create or access memory pools for custom scenarios:
+
+```csharp
+public class MyBackgroundService : BackgroundService
+{
+    private readonly MemoryPool<byte> _memoryPool;
+
+    public MyBackgroundService(IMemoryPoolFactory<byte> factory)
+    {
+        _memoryPool = factory.Create();
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                await Task.Delay(20, stoppingToken);
+                // do work that needs memory
+                var rented = _memoryPool.Rent(100);
+                rented.Dispose();
+            }
+            catch (OperationCanceledException)
+            {
+                return;
+            }
+        }
+    }
+}
+```
+
+Use a custom `IMemoryPoolFactory` to replace the memory pool being used:
+
+```csharp
+services.AddSingleton<IMemoryPoolFactory<byte>, CustomMemoryPoolFactory>();
+
+public class CustomMemoryPoolFactory : IMemoryPoolFactory<byte>
+{
+    public MemoryPool<byte> Create()
+    {
+        // Return a custom MemoryPool implementation or the default.
+        return MemoryPool<byte>.Shared;
+    }
+}
+```
+
+## Blazor WebAssembly preloading
+
+Blazor WebAssembly applications now use a `<LinkPreload />` component instead of link headers for preloading WebAssembly assets. This allows the framework to correctly identify the Blazor application base URL and provides better control over the preloading behavior.
```

Comment:
```suggestion
Blazor WebAssembly applications now use a `<LinkPreload />` component to generate `link` tags instead of link headers to preload framework assets. This allows the framework to correctly identify the Blazor application base URL and provides better control over the preloading behavior.
```